ocean_lib/models/datatoken_base.py

```diff
@@ -397,6 +397,58 @@ def build_access_service(
     def get_publish_market_order_fees(self):
         return TokenFeeInfo.from_tuple(self.contract.getPublishingMarketFee())
 
+    def buy_1dt_from_pricing_schema(self, *args, **kwargs):
```

Comment:
We should rename this to something like "get_from_pricing_schema_and_order", because, depending on conditions, at the end, we call one of:
 - buy_DT_and_order
 - dispense_and_order